substrate/frame/revive/src/lib.rs

```diff
@@ -385,118 +385,118 @@ pub mod pallet {
 	#[pallet::error]
 	pub enum Error<T> {
 		/// Invalid schedule supplied, e.g. with zero weight of a basic operation.
-		InvalidSchedule,
+		InvalidSchedule, // 0x00
```

Comment:
Instead of the comment should we have explicit enum discriminants `InvalidSchedule = 0`?

Reply:
Not sure if the macro supports it, will try it out. 

I added the comment cause that's what you get in logs / polkadotjs usually when a DispatchError is returned, 
our enum is so long that I find it convenient to have the hex code displayed next to each variant to quickly find what error was returned

Reply:
Ah yeah I see. Having in the comments would definitely help. I'm just thinking that they get easily out of sync that way as they have to be manually updated and this isn't enforced (I think with the discriminator can also easily check it if it is added for each variant explicitly).

Reply:
Yeah its super annoying when making changes that is not appending. And they **will** get out of sync and then you will be recounting anyways since you can't trust them.  But I guess its still better to have them than not. 

But really we should look into those logs where they are not decoded. That is probably an oversight and should be fixed.

Did you check whether the pallet macro supports hard coded constants? Maybe via a custom attribute?